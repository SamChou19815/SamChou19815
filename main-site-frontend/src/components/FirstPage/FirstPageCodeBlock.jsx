// @flow strict

import type { Node } from 'react';
import React from 'react';
import CodeBlock from '../Common/CodeBlock';
import styles from './FirstPageCodeBlock.module.css';

const code = `

/**
 * Copyright (C) 2015â€“2019 Developer Sam.
 *
 * @author sam
 */

// Basic information of sam, written in SAMLANG.
// The code below is a well-typed SAMLANG program.
// Try it by yourself at
// https://samlang-demo.developersam.com
 
class List<T>(Nil(unit), Cons([T * List<T>])) {
  public function <T> of(t: T): List<T> =
    Cons([t, Nil(unit)])
  public method cons(t: T): List<T> =
    Cons([t, this])
}
class Developer(
  name: string, github: string,
  projects: List<string>,
) {
  public function sam(): Developer =
    val l = List::of("SAMLANG")::cons("...");
    val github = "SamChou19815";
    { name: "Sam Zhou", github, projects: l }
}
util Main {
  function main(): Developer = Developer::sam()
}
`;

export default (): Node => <CodeBlock className={styles.Block}>{code}</CodeBlock>;
