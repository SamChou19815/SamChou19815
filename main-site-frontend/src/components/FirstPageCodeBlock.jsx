// @flow strict

import type { Node } from 'react';
import React from 'react';
import CodeBlock from './CodeBlock';
import styles from './FirstPageCodeBlock.module.css';

const code = `
/**
 * © 2015–2019 Developer Sam.
 *
 * @author sam
 */

// Basic information of sam, written in SAMLANG.
// The code below is a well-typed SAMLANG program.
// Try it by yourself at
// https://samlang-demo.developersam.com
 
class Birthday(year: int, month: int, day: int) {
  public function ofSam(): Birthday =
    { year: 1998, month: 11, day: 15 }
}
class Developer(
  name: string, university: string,
  github: string, birthday: Birthday,
) {
  public function sam(): Developer =
    val university = "Cornell University";
    val github = "SamChou19815";
    { 
      name: "Sam Zhou", university, github,
      birthday: Birthday::ofSam(),
    }
}
util Main {
  function main(): Developer = Developer::sam()
}
`;

export default (): Node => <CodeBlock className={styles.Block}>{code}</CodeBlock>;
