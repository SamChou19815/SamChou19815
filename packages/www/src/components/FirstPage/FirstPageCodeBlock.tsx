import React, { ReactElement } from 'react';

import CodeBlock from 'lib-react/PrismCodeBlock';

import styles from './FirstPageCodeBlock.module.css';

const code = `

/**
 * Copyright (C) 2015â€“2020 Developer Sam.
 *
 * @author sam
 */

// Basic information of sam, written in SAMLANG.
// The code below is a well-typed SAMLANG program.
// Try it by yourself at
// https://samlang-demo.developersam.com

class List<T>(Nil(unit), Cons([T * List<T>])) {
  function <T> of(t: T): List<T> =
    Cons([t, Nil({})])
  method cons(t: T): List<T> =
    Cons([t, this])
}
class Developer(
  val name: string, val github: string,
  val projects: List<string>,
) {
  function sam(): Developer = {
    val l = List.of("SAMLANG").cons("...")
    val github = "SamChou19815"
    { name: "Sam Zhou", github, projects: l }
  }
}
class Main {
  function main(): Developer = Developer.sam()
}
`;

export default (): ReactElement => (
  <CodeBlock language="samlang" className={styles.Block}>
    {code}
  </CodeBlock>
);
