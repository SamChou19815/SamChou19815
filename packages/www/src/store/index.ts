import { createStore, Store } from 'redux';

import { initialState as initialFileSystem } from '../filesystem';
import { FileSystemState } from '../filesystem/types';

export type TimelineState = {
  readonly workChecked: boolean;
  readonly projectsChecked: boolean;
  readonly eventsChecked: boolean;
};

export type State = {
  readonly timeline: TimelineState;
  readonly fileSystem: FileSystemState;
};

const initialState: State = {
  timeline: { workChecked: true, projectsChecked: true, eventsChecked: true },
  fileSystem: initialFileSystem
};

export type Action =
  | { readonly type: 'PATCH_TIMELINE'; readonly timeline: TimelineState }
  | { readonly type: 'PATCH_FILESYSTEM'; readonly fileSystem: FileSystemState };

const reducer = (state: State = initialState, action: Action): State => {
  switch (action.type) {
    case 'PATCH_TIMELINE':
      return { ...state, timeline: action.timeline };
    case 'PATCH_FILESYSTEM':
      return { ...state, fileSystem: action.fileSystem };
    default:
      return state;
  }
};

export type GlobalStore = Store<State, Action>;

const store: GlobalStore = createStore(reducer);
const patchTimeline = (timeline: TimelineState): Action =>
  store.dispatch({ type: 'PATCH_TIMELINE', timeline });
const patchFileSystem = (fileSystem: FileSystemState): Action =>
  store.dispatch({ type: 'PATCH_FILESYSTEM', fileSystem });

export { store, patchTimeline, patchFileSystem };
