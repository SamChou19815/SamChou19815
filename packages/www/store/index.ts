import { createStore, Store } from 'redux';

import { initialState as initialFileSystem } from '../filesystem';
import { FileSystemState } from '../filesystem/types';

export type TimelineState = {
  readonly workChecked: boolean;
  readonly projectsChecked: boolean;
  readonly eventsChecked: boolean;
};

export type State = {
  readonly theme: '' | 'dark';
  readonly timeline: TimelineState;
  readonly fileSystem: FileSystemState;
};

const initialState: State = {
  theme: '',
  timeline: { workChecked: true, projectsChecked: true, eventsChecked: true },
  fileSystem: initialFileSystem,
};

export type Action =
  | { readonly type: 'TOGGLE_THEME' }
  | { readonly type: 'PATCH_TIMELINE'; readonly timeline: TimelineState }
  | { readonly type: 'PATCH_FILESYSTEM'; readonly fileSystem: FileSystemState };

const reducer = (state: State = initialState, action: Action): State => {
  switch (action.type) {
    case 'TOGGLE_THEME': {
      const theme = state.theme === '' ? 'dark' : '';
      document.documentElement.setAttribute('data-theme', theme);
      return { ...state, theme };
    }
    case 'PATCH_TIMELINE':
      return { ...state, timeline: action.timeline };
    case 'PATCH_FILESYSTEM':
      return { ...state, fileSystem: action.fileSystem };
    default:
      return state;
  }
};

export type GlobalStore = Store<State, Action>;

const store: GlobalStore = createStore(reducer);
const toggleTheme = (): Action => store.dispatch({ type: 'TOGGLE_THEME' });
const patchTimeline = (timeline: TimelineState): Action =>
  store.dispatch({ type: 'PATCH_TIMELINE', timeline });
const patchFileSystem = (fileSystem: FileSystemState): Action =>
  store.dispatch({ type: 'PATCH_FILESYSTEM', fileSystem });

export { store, toggleTheme, patchTimeline, patchFileSystem };
