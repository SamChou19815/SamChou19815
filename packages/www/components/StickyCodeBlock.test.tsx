import React from 'react';

import { Provider as ReactReduxProvider } from 'react-redux';
import renderer from 'react-test-renderer';

import { store } from '../store';
import FirstPageCodeBlock from './StickyCodeBlock';

it('FirstPageCodeBlock matches snapshot.', () => {
  const tree = renderer
    .create(
      <ReactReduxProvider store={store}>
        <FirstPageCodeBlock />
      </ReactReduxProvider>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
