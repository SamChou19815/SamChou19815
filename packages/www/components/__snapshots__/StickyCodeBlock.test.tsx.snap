// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FirstPageCodeBlock matches snapshot. 1`] = `
<pre
  className="prism-code language-samlang Block"
  style={
    Object {
      "backgroundColor": "var(--ifm-background-color)",
    }
  }
>
  <div
    className="token-line"
    style={
      Object {
        "backgroundColor": null,
      }
    }
  >
    <span
      className="token comment"
      style={
        Object {
          "color": "rgb(128, 128, 128)",
          "fontStyle": "italic",
        }
      }
    >
      /**
    </span>
  </div>
  <div
    className="token-line"
    style={
      Object {
        "backgroundColor": null,
      }
    }
  >
    <span
      className="token comment"
      style={
        Object {
          "color": "rgb(128, 128, 128)",
          "fontStyle": "italic",
        }
      }
    >
       * Copyright (C) 2015â€“2020 Developer Sam.
    </span>
  </div>
  <div
    className="token-line"
    style={
      Object {
        "backgroundColor": null,
      }
    }
  >
    <span
      className="token comment"
      style={
        Object {
          "color": "rgb(128, 128, 128)",
          "fontStyle": "italic",
        }
      }
    >
       *
    </span>
  </div>
  <div
    className="token-line"
    style={
      Object {
        "backgroundColor": null,
      }
    }
  >
    <span
      className="token comment"
      style={
        Object {
          "color": "rgb(128, 128, 128)",
          "fontStyle": "italic",
        }
      }
    >
       * @author sam
    </span>
  </div>
  <div
    className="token-line"
    style={
      Object {
        "backgroundColor": null,
      }
    }
  >
    <span
      className="token comment"
      style={
        Object {
          "color": "rgb(128, 128, 128)",
          "fontStyle": "italic",
        }
      }
    >
       */
    </span>
    <span
      className="token plain"
    >
      
    </span>
  </div>
  <div
    className="token-line"
    style={
      Object {
        "backgroundColor": null,
      }
    }
  >
    <span
      className="token plain"
      style={
        Object {
          "display": "inline-block",
        }
      }
    >
      
    </span>
  </div>
  <div
    className="token-line"
    style={
      Object {
        "backgroundColor": null,
      }
    }
  >
    <span
      className="token plain"
    >
      
    </span>
    <span
      className="token comment"
      style={
        Object {
          "color": "rgb(128, 128, 128)",
          "fontStyle": "italic",
        }
      }
    >
      // Basic information of sam, written in SAMLANG.
    </span>
    <span
      className="token plain"
    >
      
    </span>
  </div>
  <div
    className="token-line"
    style={
      Object {
        "backgroundColor": null,
      }
    }
  >
    <span
      className="token plain"
    >
      
    </span>
    <span
      className="token comment"
      style={
        Object {
          "color": "rgb(128, 128, 128)",
          "fontStyle": "italic",
        }
      }
    >
      // The code below is a well-typed SAMLANG program.
    </span>
    <span
      className="token plain"
    >
      
    </span>
  </div>
  <div
    className="token-line"
    style={
      Object {
        "backgroundColor": null,
      }
    }
  >
    <span
      className="token plain"
    >
      
    </span>
    <span
      className="token comment"
      style={
        Object {
          "color": "rgb(128, 128, 128)",
          "fontStyle": "italic",
        }
      }
    >
      // Try it by yourself at
    </span>
    <span
      className="token plain"
    >
      
    </span>
  </div>
  <div
    className="token-line"
    style={
      Object {
        "backgroundColor": null,
      }
    }
  >
    <span
      className="token plain"
    >
      
    </span>
    <span
      className="token comment"
      style={
        Object {
          "color": "rgb(128, 128, 128)",
          "fontStyle": "italic",
        }
      }
    >
      // https://samlang.developersam.com/demo
    </span>
    <span
      className="token plain"
    >
      
    </span>
  </div>
  <div
    className="token-line"
    style={
      Object {
        "backgroundColor": null,
      }
    }
  >
    <span
      className="token plain"
      style={
        Object {
          "display": "inline-block",
        }
      }
    >
      
    </span>
  </div>
  <div
    className="token-line"
    style={
      Object {
        "backgroundColor": null,
      }
    }
  >
    <span
      className="token plain"
    >
      
    </span>
    <span
      className="token keyword"
      style={
        Object {
          "color": "rgb(62, 122, 226)",
        }
      }
    >
      class
    </span>
    <span
      className="token plain"
    >
       
    </span>
    <span
      className="token class-name"
      style={
        Object {
          "color": "rgb(154, 48, 173)",
        }
      }
    >
      List
    </span>
    <span
      className="token plain"
    >
      &lt;
    </span>
    <span
      className="token class-name"
      style={
        Object {
          "color": "rgb(154, 48, 173)",
        }
      }
    >
      T
    </span>
    <span
      className="token plain"
    >
      &gt;(
    </span>
    <span
      className="token class-name"
      style={
        Object {
          "color": "rgb(154, 48, 173)",
        }
      }
    >
      Nil
    </span>
    <span
      className="token plain"
    >
      (
    </span>
    <span
      className="token keyword"
      style={
        Object {
          "color": "rgb(62, 122, 226)",
        }
      }
    >
      unit
    </span>
    <span
      className="token plain"
    >
      ), 
    </span>
    <span
      className="token class-name"
      style={
        Object {
          "color": "rgb(154, 48, 173)",
        }
      }
    >
      Cons
    </span>
    <span
      className="token plain"
    >
      ([
    </span>
    <span
      className="token class-name"
      style={
        Object {
          "color": "rgb(154, 48, 173)",
        }
      }
    >
      T
    </span>
    <span
      className="token plain"
    >
       
    </span>
    <span
      className="token operator"
      style={
        Object {
          "color": "rgb(56, 72, 79)",
        }
      }
    >
      *
    </span>
    <span
      className="token plain"
    >
       
    </span>
    <span
      className="token class-name"
      style={
        Object {
          "color": "rgb(154, 48, 173)",
        }
      }
    >
      List
    </span>
    <span
      className="token plain"
    >
      &lt;
    </span>
    <span
      className="token class-name"
      style={
        Object {
          "color": "rgb(154, 48, 173)",
        }
      }
    >
      T
    </span>
    <span
      className="token plain"
    >
      &gt;])) {
    </span>
  </div>
  <div
    className="token-line"
    style={
      Object {
        "backgroundColor": null,
      }
    }
  >
    <span
      className="token plain"
    >
        
    </span>
    <span
      className="token keyword"
      style={
        Object {
          "color": "rgb(62, 122, 226)",
        }
      }
    >
      function
    </span>
    <span
      className="token plain"
    >
       &lt;
    </span>
    <span
      className="token class-name"
      style={
        Object {
          "color": "rgb(154, 48, 173)",
        }
      }
    >
      T
    </span>
    <span
      className="token plain"
    >
      &gt; 
    </span>
    <span
      className="token function"
      style={
        Object {
          "color": "rgb(213, 34, 98)",
        }
      }
    >
      of
    </span>
    <span
      className="token plain"
    >
      (t: 
    </span>
    <span
      className="token class-name"
      style={
        Object {
          "color": "rgb(154, 48, 173)",
        }
      }
    >
      T
    </span>
    <span
      className="token plain"
    >
      ): 
    </span>
    <span
      className="token class-name"
      style={
        Object {
          "color": "rgb(154, 48, 173)",
        }
      }
    >
      List
    </span>
    <span
      className="token plain"
    >
      &lt;
    </span>
    <span
      className="token class-name"
      style={
        Object {
          "color": "rgb(154, 48, 173)",
        }
      }
    >
      T
    </span>
    <span
      className="token plain"
    >
      &gt; =
    </span>
  </div>
  <div
    className="token-line"
    style={
      Object {
        "backgroundColor": null,
      }
    }
  >
    <span
      className="token plain"
    >
          
    </span>
    <span
      className="token class-name"
      style={
        Object {
          "color": "rgb(154, 48, 173)",
        }
      }
    >
      Cons
    </span>
    <span
      className="token plain"
    >
      ([t, 
    </span>
    <span
      className="token class-name"
      style={
        Object {
          "color": "rgb(154, 48, 173)",
        }
      }
    >
      Nil
    </span>
    <span
      className="token plain"
    >
      ({})])
    </span>
  </div>
  <div
    className="token-line"
    style={
      Object {
        "backgroundColor": null,
      }
    }
  >
    <span
      className="token plain"
    >
        
    </span>
    <span
      className="token keyword"
      style={
        Object {
          "color": "rgb(62, 122, 226)",
        }
      }
    >
      method
    </span>
    <span
      className="token plain"
    >
       
    </span>
    <span
      className="token function"
      style={
        Object {
          "color": "rgb(213, 34, 98)",
        }
      }
    >
      cons
    </span>
    <span
      className="token plain"
    >
      (t: 
    </span>
    <span
      className="token class-name"
      style={
        Object {
          "color": "rgb(154, 48, 173)",
        }
      }
    >
      T
    </span>
    <span
      className="token plain"
    >
      ): 
    </span>
    <span
      className="token class-name"
      style={
        Object {
          "color": "rgb(154, 48, 173)",
        }
      }
    >
      List
    </span>
    <span
      className="token plain"
    >
      &lt;
    </span>
    <span
      className="token class-name"
      style={
        Object {
          "color": "rgb(154, 48, 173)",
        }
      }
    >
      T
    </span>
    <span
      className="token plain"
    >
      &gt; =
    </span>
  </div>
  <div
    className="token-line"
    style={
      Object {
        "backgroundColor": null,
      }
    }
  >
    <span
      className="token plain"
    >
          
    </span>
    <span
      className="token class-name"
      style={
        Object {
          "color": "rgb(154, 48, 173)",
        }
      }
    >
      Cons
    </span>
    <span
      className="token plain"
    >
      ([t, 
    </span>
    <span
      className="token keyword"
      style={
        Object {
          "color": "rgb(62, 122, 226)",
        }
      }
    >
      this
    </span>
    <span
      className="token plain"
    >
      ])
    </span>
  </div>
  <div
    className="token-line"
    style={
      Object {
        "backgroundColor": null,
      }
    }
  >
    <span
      className="token plain"
    >
      }
    </span>
  </div>
  <div
    className="token-line"
    style={
      Object {
        "backgroundColor": null,
      }
    }
  >
    <span
      className="token plain"
    >
      
    </span>
    <span
      className="token keyword"
      style={
        Object {
          "color": "rgb(62, 122, 226)",
        }
      }
    >
      class
    </span>
    <span
      className="token plain"
    >
       
    </span>
    <span
      className="token class-name"
      style={
        Object {
          "color": "rgb(154, 48, 173)",
        }
      }
    >
      Developer
    </span>
    <span
      className="token plain"
    >
      (
    </span>
  </div>
  <div
    className="token-line"
    style={
      Object {
        "backgroundColor": null,
      }
    }
  >
    <span
      className="token plain"
    >
        
    </span>
    <span
      className="token keyword"
      style={
        Object {
          "color": "rgb(62, 122, 226)",
        }
      }
    >
      val
    </span>
    <span
      className="token plain"
    >
       name: 
    </span>
    <span
      className="token keyword"
      style={
        Object {
          "color": "rgb(62, 122, 226)",
        }
      }
    >
      string
    </span>
    <span
      className="token plain"
    >
      , 
    </span>
    <span
      className="token keyword"
      style={
        Object {
          "color": "rgb(62, 122, 226)",
        }
      }
    >
      val
    </span>
    <span
      className="token plain"
    >
       github: 
    </span>
    <span
      className="token keyword"
      style={
        Object {
          "color": "rgb(62, 122, 226)",
        }
      }
    >
      string
    </span>
    <span
      className="token plain"
    >
      ,
    </span>
  </div>
  <div
    className="token-line"
    style={
      Object {
        "backgroundColor": null,
      }
    }
  >
    <span
      className="token plain"
    >
        
    </span>
    <span
      className="token keyword"
      style={
        Object {
          "color": "rgb(62, 122, 226)",
        }
      }
    >
      val
    </span>
    <span
      className="token plain"
    >
       projects: 
    </span>
    <span
      className="token class-name"
      style={
        Object {
          "color": "rgb(154, 48, 173)",
        }
      }
    >
      List
    </span>
    <span
      className="token plain"
    >
      &lt;
    </span>
    <span
      className="token keyword"
      style={
        Object {
          "color": "rgb(62, 122, 226)",
        }
      }
    >
      string
    </span>
    <span
      className="token plain"
    >
      &gt;,
    </span>
  </div>
  <div
    className="token-line"
    style={
      Object {
        "backgroundColor": null,
      }
    }
  >
    <span
      className="token plain"
    >
      ) {
    </span>
  </div>
  <div
    className="token-line"
    style={
      Object {
        "backgroundColor": null,
      }
    }
  >
    <span
      className="token plain"
    >
        
    </span>
    <span
      className="token keyword"
      style={
        Object {
          "color": "rgb(62, 122, 226)",
        }
      }
    >
      function
    </span>
    <span
      className="token plain"
    >
       
    </span>
    <span
      className="token function"
      style={
        Object {
          "color": "rgb(213, 34, 98)",
        }
      }
    >
      sam
    </span>
    <span
      className="token plain"
    >
      (): 
    </span>
    <span
      className="token class-name"
      style={
        Object {
          "color": "rgb(154, 48, 173)",
        }
      }
    >
      Developer
    </span>
    <span
      className="token plain"
    >
       = {
    </span>
  </div>
  <div
    className="token-line"
    style={
      Object {
        "backgroundColor": null,
      }
    }
  >
    <span
      className="token plain"
    >
          
    </span>
    <span
      className="token keyword"
      style={
        Object {
          "color": "rgb(62, 122, 226)",
        }
      }
    >
      val
    </span>
    <span
      className="token plain"
    >
       l = 
    </span>
    <span
      className="token class-name"
      style={
        Object {
          "color": "rgb(154, 48, 173)",
        }
      }
    >
      List
    </span>
    <span
      className="token plain"
    >
      .
    </span>
    <span
      className="token function"
      style={
        Object {
          "color": "rgb(213, 34, 98)",
        }
      }
    >
      of
    </span>
    <span
      className="token plain"
    >
      (
    </span>
    <span
      className="token string"
      style={
        Object {
          "color": "rgb(26, 143, 82)",
        }
      }
    >
      "SAMLANG"
    </span>
    <span
      className="token plain"
    >
      ).
    </span>
    <span
      className="token function"
      style={
        Object {
          "color": "rgb(213, 34, 98)",
        }
      }
    >
      cons
    </span>
    <span
      className="token plain"
    >
      (
    </span>
    <span
      className="token string"
      style={
        Object {
          "color": "rgb(26, 143, 82)",
        }
      }
    >
      "..."
    </span>
    <span
      className="token plain"
    >
      )
    </span>
  </div>
  <div
    className="token-line"
    style={
      Object {
        "backgroundColor": null,
      }
    }
  >
    <span
      className="token plain"
    >
          
    </span>
    <span
      className="token keyword"
      style={
        Object {
          "color": "rgb(62, 122, 226)",
        }
      }
    >
      val
    </span>
    <span
      className="token plain"
    >
       github = 
    </span>
    <span
      className="token string"
      style={
        Object {
          "color": "rgb(26, 143, 82)",
        }
      }
    >
      "SamChou19815"
    </span>
    <span
      className="token plain"
    >
      
    </span>
  </div>
  <div
    className="token-line"
    style={
      Object {
        "backgroundColor": null,
      }
    }
  >
    <span
      className="token plain"
    >
          { name: 
    </span>
    <span
      className="token string"
      style={
        Object {
          "color": "rgb(26, 143, 82)",
        }
      }
    >
      "Sam Zhou"
    </span>
    <span
      className="token plain"
    >
      , github, projects: l }
    </span>
  </div>
  <div
    className="token-line"
    style={
      Object {
        "backgroundColor": null,
      }
    }
  >
    <span
      className="token plain"
    >
        }
    </span>
  </div>
  <div
    className="token-line"
    style={
      Object {
        "backgroundColor": null,
      }
    }
  >
    <span
      className="token plain"
    >
      }
    </span>
  </div>
  <div
    className="token-line"
    style={
      Object {
        "backgroundColor": null,
      }
    }
  >
    <span
      className="token plain"
    >
      
    </span>
    <span
      className="token keyword"
      style={
        Object {
          "color": "rgb(62, 122, 226)",
        }
      }
    >
      class
    </span>
    <span
      className="token plain"
    >
       
    </span>
    <span
      className="token class-name"
      style={
        Object {
          "color": "rgb(154, 48, 173)",
        }
      }
    >
      Main
    </span>
    <span
      className="token plain"
    >
       {
    </span>
  </div>
  <div
    className="token-line"
    style={
      Object {
        "backgroundColor": null,
      }
    }
  >
    <span
      className="token plain"
    >
        
    </span>
    <span
      className="token keyword"
      style={
        Object {
          "color": "rgb(62, 122, 226)",
        }
      }
    >
      function
    </span>
    <span
      className="token plain"
    >
       
    </span>
    <span
      className="token function"
      style={
        Object {
          "color": "rgb(213, 34, 98)",
        }
      }
    >
      main
    </span>
    <span
      className="token plain"
    >
      (): 
    </span>
    <span
      className="token class-name"
      style={
        Object {
          "color": "rgb(154, 48, 173)",
        }
      }
    >
      Developer
    </span>
    <span
      className="token plain"
    >
       = 
    </span>
    <span
      className="token class-name"
      style={
        Object {
          "color": "rgb(154, 48, 173)",
        }
      }
    >
      Developer
    </span>
    <span
      className="token plain"
    >
      .
    </span>
    <span
      className="token function"
      style={
        Object {
          "color": "rgb(213, 34, 98)",
        }
      }
    >
      sam
    </span>
    <span
      className="token plain"
    >
      ()
    </span>
  </div>
  <div
    className="token-line"
    style={
      Object {
        "backgroundColor": null,
      }
    }
  >
    <span
      className="token plain"
    >
      }
    </span>
  </div>
</pre>
`;
